src/interpreter.rs

```diff
@@ -239,8 +239,8 @@ pub fn execute_program(prog_: Option<&[u8]>, mem: &[u8], mbuff: &[u8], helpers:
             ebpf::XOR32_REG  =>   reg[_dst] = (reg[_dst] as u32             ^ reg[_src] as u32) as u64,
             ebpf::MOV32_IMM  =>   reg[_dst] = insn.imm   as u32                                 as u64,
             ebpf::MOV32_REG  =>   reg[_dst] = (reg[_src] as u32)                                as u64,
-            ebpf::ARSH32_IMM => { reg[_dst] = (reg[_dst] as i32).wrapping_shr(insn.imm  as u32) as u64; reg[_dst] &= U32MAX; },
-            ebpf::ARSH32_REG => { reg[_dst] = (reg[_dst] as i32).wrapping_shr(reg[_src] as u32) as u64; reg[_dst] &= U32MAX; },
+            ebpf::ARSH32_IMM => { reg[_dst] = (reg[_dst] as i32).wrapping_shr(insn.imm  as u32 & SHIFT_MASK_32) as u64; reg[_dst] &= U32MAX; },
+            ebpf::ARSH32_REG => { reg[_dst] = (reg[_dst] as i32).wrapping_shr(reg[_src] as u32 & SHIFT_MASK_32) as u64; reg[_dst] &= U32MAX; },
```

Comment:
Did you check that this part is necessary? [I wrote some tests](https://github.com/qmonnet/rbpf/tree/arsh-fix-tests) and the 32-bit versions pass successfully without this change. I suspect that the `.wrapping_shr()` does something similar already? If that's the case, maybe just add a comment to explain why we don't need additional masking?

Reply:
Many thanks! This mask is unnecessary since the wrapper_shr has already been implemented as `self >> mask(rhs)` according to the rust docs.